### PR TITLE
Blast fails with too many files open exception

### PR DESCRIPTION
**Pull request by eparejatobes**
https://github.com/ohnosequences/metagenomica/pull/48 followup.

This problem is still present. If you run blast step for too long, some workers start failing with "Too many files open" exception.

### CODE REVIEW

File: src/main/scala/metagenomica/loquats/3.blast.scala [hunk 1]
```diff
@@ -46,14 +46,13 @@ extends DataProcessingBundle(
...
+            fasta.sequence(FastaSequence(quartet(1)))    ::
             *[AnyDenotation]
           )
 
         val readFile = context / "read.fa"
-        readFile
-          .createIfNotExists()
-          .appendLines(read.toLines: _*)
+
+        readFile.append(read.toLines)
```
Comment (laughedelic) [inline src/main/scala/metagenomica/loquats/3.blast.scala]: I think you can use `.overwrite` here and then you don't need to `.clear` it.

**Comment by laughedelic** [comment]
This version is running (with the `M2-blast-bugfix` suffix). So far it goes well.

**Comment by eparejatobes** [comment]
good. Can I log into the instances somehow?

**Comment by laughedelic** [comment]
I don't think so. It's on my ssh-key. But you can check the logs in `s3://loquat.testing/blast-mg7-1-0-0-m2-blast-bugfix-snapshot/`

**Comment by eparejatobes** [comment]
OK thanks, we'll see tomorrow anyway

**Comment by eparejatobes** [comment]
as a note, looks like this: https://github.com/ohnosequences/mg7/blob/bug/blast/open/files/26/src/main/scala/metagenomica/loquats/3.blast.scala#L83 adds an empty line per batch. The results are (for `s3://resources.ohnosequences.com/16s/public-datasets/PRJEB6592/blast-test/ERR567374/blast.547.csv` for example)

``` csv
ERR567374.4439,447,5,447,URS0000881CD7,1397,356,798,840,0
ERR567374.4439,447,5,447,URS00007F9683,1023,269,711,840,0
ERR567374.4439,447,5,447,URS000069491C,1395,354,796,840,0
ERR567374.4439,447,5,447,URS0000601BB6,593,85,527,840,0
ERR567374.4439,447,5,447,URS000059F170,1432,319,761,840,0
ERR567374.4439,447,5,447,URS000053FEBA,1353,327,769,840,0
ERR567374.4439,447,5,447,URS0000530672,901,338,780,840,0
ERR567374.4439,447,5,447,URS00004DB2E2,1350,327,769,840,0
ERR567374.4439,447,5,447,URS0000434C43,1374,337,779,840,0
ERR567374.4439,447,5,447,URS000041DF71,810,354,796,840,0

ERR567374.4440,444,5,436,URS00008A9D3F,1347,318,749,796,0
ERR567374.4440,444,5,436,URS00008A4D13,1341,317,748,796,0
ERR567374.4440,444,5,436,URS0000898159,1355,332,763,796,0
ERR567374.4440,444,5,436,URS0000892E20,1351,332,763,796,0
ERR567374.4440,444,5,436,URS000088984B,1349,325,756,796,0
ERR567374.4440,444,5,436,URS0000885FFA,1351,327,758,796,0
ERR567374.4440,444,5,436,URS000087F0BB,1347,318,749,796,0
ERR567374.4440,444,5,436,URS000087C0A0,1341,318,749,796,0
ERR567374.4440,444,5,436,URS0000878F2D,1353,318,749,796,0
ERR567374.4440,444,5,436,URS00008721B8,1339,315,746,796,0

ERR567374.4441,447,5,447,URS0000881CD7,1397,356,798,846,0
ERR567374.4441,447,5,447,URS00007F9683,1023,269,711,846,0
ERR567374.4441,447,5,447,URS000069491C,1395,354,796,846,0
ERR567374.4441,447,5,447,URS0000601BB6,593,85,527,846,0
ERR567374.4441,447,5,447,URS000059F170,1432,319,761,846,0
ERR567374.4441,447,5,447,URS000053FEBA,1353,327,769,846,0
ERR567374.4441,447,5,447,URS0000530672,901,338,780,846,0
ERR567374.4441,447,5,447,URS00004DB2E2,1350,327,769,846,0
ERR567374.4441,447,5,447,URS0000434C43,1374,337,779,846,0
ERR567374.4441,447,5,447,URS000041DF71,810,354,796,846,0
```

**Comment by eparejatobes** [comment]
Just a quick note: this looks to be caused by loquat, where the sandbox folders are deleted an created at every step, and somehow this leads to a file descriptor leakage.

**Comment by laughedelic** [comment]
Opened https://github.com/ohnosequences/loquat/issues/56 for solving this.

**Comment by laughedelic** [comment]
@eparejatobes btw, about the extra newline, it's not about `<<`, it's the way FASTA `.toLines` works.

**Pull request merged**